### PR TITLE
[doc] How to adapt test cases for SONiC Chassis.

### PR DESCRIPTION
**Pull request by sanmalho-git**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
Need a document to outline how the enhancements made to sonic-mgmt can be brought together to adapt existing OC tests to work against a SONiC Chassis.

#### How did you do it?
Added md document

### CODE REVIEW

**Comment by shubav** [comment]
Could we change the filename to pytest.voq_chassis.md in line with the current convention in that dir?

File: tests/docs/pytest.voq_chassis.md [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+This document describes the design details for testing SONiC Chassis by extending the existing sonic-mgmt test framework.  
+
+# Scope
+A SONiC Chassis typically has multiple line cards and supervisor card. Today, there is no 'single point of mgmt' for a SONiC Chassis. In a dis-aggregated model, each linecard
+and supervisor card has its own management IP address, that can be controlled/configured independently. 
+
+The existing sonic-mgmt test framework has well defined test methodologies for testing of SONiC pizza boxes (a single mgmt. IP). 
+The scope of this document is to enhance the framework to test SONiC Chassis with multiple mgmt. IP addresses, while still utilizing the existing test methodologies. 
+
+The end goal is to be able to run the existing OC tests against a SONiC Chassis with minimal changes to test cases itself.
```
Comment (wangxin) [inline tests/docs/pytest.voq_chassis.md:45]: What is `OC`?

File: tests/docs/pytest.voq_chassis.md [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+The ansible modules that are applicable to be run in different namespaces have been enhanced to support such functionaliy. Currently, the following modules support running them on different namespaces:
+- bgp_facts - added 'instance_id' as an optional module param.
+- config_facts - added 'namespace' as an opitonal module param.
+
+If the optional namespace/ASIC specific module params are not specified, then they default to returning the output from the global namespace.
+
+Other modules will be enhanced as needed during adaption of the tests to SONiC chassis.
+
+## 2.2 Parameterizing dut_index and asic_index
```
Comment (wangxin) [inline tests/docs/pytest.voq_chassis.md:74]: The infrastructure has been extended to support below fixtures for parameterizing test cases:
* enum_dut_index
* enum_dut_hostname
* enum_asic_index
* enum_frontend_asic_index
* enum_dut_portname
* enum_dut_portname_oper_up
* enum_dut_portname_admin_up
* enum_dut_portchannel
* enum_dut_portchannel_oper_up
* enum_dut_portchannel_admin_up
* enum_dut_feature
Please update this document accordingly.

File: tests/docs/pytest.voq_chassis.md [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+Below, we take the example test_bgp_facts modified with both approaches as an example.
+
+## 5.1 parameterizing dut_index
+
+- Modify definition to replace 'duthost' with 'duthosts', and include 'dut_index' and 'asic_index'.
+- Get duthost based on dut_index from duthosts
+- Call ansible modules 'bgp_facts' and config_facts' using the namespace arguements. 
+
+<pre>
+def test_bgp_facts(<b>duthosts, dut_index, asic_index</b>):
```
Comment (wangxin) [inline tests/docs/pytest.voq_chassis.md:231]: Fixtures like `dut_index` and `asic_index` have been updated to `enum_dut_hostname` and `enum_asic_index`. Please update occurrences of the deprecated fixture names accordingly.

**Review by wangxin — CHANGES_REQUESTED** [review]

**Pull request merged**